sdk/servicebus/azure-servicebus/azure/servicebus/amqp/_amqp_utils.py

```diff
@@ -0,0 +1,25 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# -------------------------------------------------------------------------
+
+def encode_str(data, encoding='utf-8'):
+    try:
+        return data.encode(encoding)
+    except AttributeError:
+        return data
+
+def normalized_data_body(data, **kwargs):
+    # A helper method to normalize input into AMQP Data Body format
+    encoding = kwargs.get("encoding", "utf-8")
+    if isinstance(data, list):
+        return [encode_str(item, encoding) for item in data]
+    return [encode_str(data, encoding)]
+
+def normalized_sequence_body(sequence):
+    # A helper method to normalize input into AMQP Sequence Body format
+    if isinstance(sequence, list) and all([isinstance(b, list) for b in sequence]):
+        return sequence
+    if isinstance(sequence, list):
+        return [sequence]
```

Comment:
It seems that when we're encoding a sequence, we iterate through a list of items. https://github.com/Azure/azure-sdk-for-python/blob/b808884163b709143a678d281540175b42deb5c0/sdk/servicebus/azure-servicebus/azure/servicebus/_pyamqp/_encode.py#L848

So I think this normalized like this so that we can use encode items in a loop, even if it's just one item.

Reply:
Yes that explains one of the lists, just not why it needs to be wrapped in a second one...

Reply:
so just tested sending difference seq bodies:
- [b'message', 123.456, True]
- [b'message', [123.456, ['hi', 'bye']], True]

And it looks like this for loop actually isn't needed. We can remove both this loop and the "normalize_sequence_body" bits and this send and receives the same.

Reply:
created an issue for this here: #29701